adam-core/src/main/scala/org/bdgenomics/adam/util/IndexedFastaFile.scala

```diff
@@ -0,0 +1,82 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bdgenomics.adam.util
+
+import java.net.URI
+import java.nio.file.Paths
+
+import htsjdk.samtools.reference.{ FastaSequenceIndex, IndexedFastaSequenceFile }
+import org.apache.hadoop.fs.Path
+import org.apache.spark.SparkContext
+import org.bdgenomics.adam.models.{ SequenceDictionary, ReferenceRegion }
+import org.bdgenomics.utils.misc.Logging
+import picard.sam.CreateSequenceDictionary
+
+/**
+ * Loads and extracts sequences directly from indexed fasta or fa files. filePath requires fai index in the
+ * same directory with same naming convention.
+ *
+ * @param filePath path to fasta or fa index
+ */
+case class IndexedFastaFile(sc: SparkContext, filePath: String) extends ReferenceFile with Logging {
+
+  // get absolute path and scheme to create URI
+  val path = new Path(filePath).toString
+  val scheme: String = new Path(path).getFileSystem(sc.hadoopConfiguration).getScheme
+  val pathWithScheme = s"${scheme}://${path}"
+
+  val uri: URI = new URI(pathWithScheme)
+
+  val file = Paths.get(uri).toFile
+
+  // Generate IndexedFastaSequenceFile from path and fai index
+  val ref: IndexedFastaSequenceFile = {
+    val uriIdx: URI = new URI(pathWithScheme + ".fai")
+    val pathIdx = Paths.get(uriIdx)
+
+    val idx = new FastaSequenceIndex(pathIdx.toFile)
+    new IndexedFastaSequenceFile(file, idx)
+  }
+
+  // Get sequence dictionary. If sequence dictionary is not defined,
+  // generate sequence dictionary from dile
+  val sequences =
+    try {
+      SequenceDictionary(ref.getSequenceDictionary)
+    } catch {
+      case e: Exception => {
```

Comment:
In that case, you'd pass `stringency = ValidationStringency.STRICT` from Mango. There's several reasons (in no specific order) I'd like to fail(/validate) instead of regenerate the `.dict` file:

1. Consistency with other tools. All of the Picard CLI's (and the GATK) will fail by default if the `.dict` file is missing. They won't regenerate the file.
2. Simplicity of handling edge cases. If loading the sequence dictionary fails, this could've happened because the file is missing (and then regenerating the file is a reasonable answer) but it could also be a file permission issue, corrupt data on file system, bad Hadoop configuration causes JSR-203-hadoop to barf, etc. In all of those cases, regenerating the sequence dictionary is not the correct answer.
3. Packaging. I don't want to add Picard as a dependency.